packages/web/e2e/page-object-models/socialActions.ts

```diff
@@ -0,0 +1,74 @@
+import { Locator, Page, expect } from '@playwright/test'
+
+export class SocialActions {
+  public readonly favoriteButton: Locator
+  public readonly unfavoriteButton: Locator
+  public readonly repostButton: Locator
+  public readonly unrepostButton: Locator
+
+  constructor(page: Page) {
+    const trackActions = page.getByRole('group', {
+      name: /track actions/i
+    })
+    this.favoriteButton = trackActions.getByRole('button', {
+      name: /favorite$/i
+    })
+    this.unfavoriteButton = trackActions.getByRole('button', {
+      name: /favorited$/i
+    })
+    this.repostButton = trackActions.getByRole('button', {
+      name: /repost$/i
+    })
+    this.unrepostButton = trackActions.getByRole('button', {
+      name: /reposted$/i
+    })
+  }
+
+  async favorite() {
+    // Unfavorited => Favorited
+    await this.favoriteButton.click()
+
+    // Wait for indexing, reload
+    await new Promise((resolve) => setTimeout(resolve, 1000))
```

Comment:
PW has a built in wait for timeout
```
await page.waitForTimeout(1000)
```

Reply:
That won't work for us - that only waits for timeout _on the navigation_, we need it to wait for the hydration of the page

Reply:
yeah I just realized this is not the spot for it 💀

Reply:
no you're right here - I can use that here!

Reply:
oh yeah, but I don't have `page` in scope 💀